pvlib/pvsystem.py

```diff
@@ -400,7 +406,9 @@ def __init__(self, pvsystem=None, location=None, **kwargs):
 
         # get and combine attributes from the pvsystem and/or location
         # with the rest of the kwargs
-
+        
+        self.location = location
```

Comment:
I think I know why you're adding this ``location`` attribute (to avoid copy/paste location representation code), but I don't think we should do it. The input ``location``'s attributes are reassigned as attributes of the new object. Adding a separate ``location`` attribute duplicates the information and potentially causes confusion. Also, changes to the localized object's location-like attributes would not be reflected in the ``__repr__`` return.